plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +128,19 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        # Check whether getGlobalContainerStack() returns None or not...
+        global_container_stack = Application.getInstance().getGlobalContainerStack()
+        if not global_container_stack:
+            Logger.log("c", "getGlobalContainerStack() returned None")
+            Logger.log("i", "Closing firmware-updater UI as a consequence")
```

Comment:
I don't think this warrants a _critical_ error. A critical error means the application cannot (should not) continue. This hypothetical case means the function cannot complete. Also I don't think it is necessary to create 2 log entries. I suggest ```Logger.log("e", "There is no global container stack. Can not update firmware")```